Sources/MaliciousSiteProtection/MaliciousSiteDetector.swift

```diff
@@ -104,27 +83,29 @@ public class PhishingDetector: PhishingDetecting {
         let matches = await fetchMatches(hashPrefix: hashPrefixParam)
         let hostnameHash = generateHashPrefix(for: canonicalHost, length: Int.max)
         for match in matches where matchesUrl(hash: match.hash, regexPattern: match.regex, url: canonicalUrl, hostnameHash: hostnameHash) {
-            eventMapping.fire(PhishingDetectionEvents.errorPageShown(clientSideHit: false))
+            eventMapping.fire(.errorPageShown(clientSideHit: false))
             return true
         }
         return false
     }
 
-    public func isMalicious(url: URL) async -> Bool {
-        guard let canonicalHost = url.canonicalHost(), let canonicalUrl = url.canonicalURL() else { return false }
-
-        let hashPrefix = generateHashPrefix(for: canonicalHost, length: hashPrefixStoreLength)
-        if dataStore.hashPrefixes.contains(hashPrefix) {
-            // Check local filterSet first
-            if checkLocalFilters(canonicalHost: canonicalHost, canonicalUrl: canonicalUrl) {
-                return true
-            }
-            // If nothing found, hit the API to get matches
-            if await checkApiMatches(canonicalHost: canonicalHost, canonicalUrl: canonicalUrl) {
-                return true
+    public func evaluate(_ url: URL) async -> ThreatKind? {
```

Comment:
I think this is fine for now, but soon this will result in two API requests when we introduce malware protections. It also doesn't allow for independent feature flagging. We should discuss if/how we want to do this on Monday in our call. Doesn't block this PR though.

Reply:
good, let‘s mark this as a discussion point, I was assuming the API would check for both threat kinds and return the detected kind if any;